app/server/lib/ActiveDoc.ts

```diff
@@ -2614,6 +2623,44 @@ export class ActiveDoc extends EventEmitter {
     }];
   }
 
+  private async _updateAttachmentFileSizesUsingFileIdent(docSession: OptDocSession,
+                                                         newFileSizesByFileIdent: Map<string, number>): Promise<void> {
+    const rowIdsToUpdate: number[] = [];
+    const newFileSizesForRows: CellValue[] = [];
+    const attachments = this.docData?.getMetaTable("_grist_Attachments").getRecords();
+    for (const attachmentRec of attachments ?? []) {
+      const newSize = newFileSizesByFileIdent.get(attachmentRec.fileIdent);
+      if (newSize && newSize !== attachmentRec.fileSize) {
+        rowIdsToUpdate.push(attachmentRec.id);
+        newFileSizesForRows.push(newSize);
+      }
+    }
+
+    if (rowIdsToUpdate.length === 0) {
+      return;
+    }
+
+    const action: BulkUpdateRecord = ['BulkUpdateRecord', '_grist_Attachments', rowIdsToUpdate, {
+      fileSize: newFileSizesForRows
+    }];
+
+    try {
+      await this._applyUserActionsWithExtendedOptions(
+        docSession,
+        [action],
+        {attachment: true},
+      );
+    } catch (e) {
+      if (e instanceof SandboxError) {
+        this._log.error(null, "Attachment sizes could not be updated due to a sandbox error: ", e);
+        throw new Error("Attachment sizes could not be updated due to a sandbox error", { cause: e });
+      }
+    }
```

Comment:
```suggestion
     throw e;
    }
```

Should we rethrow other errors?